src/stm32f0xx/gpio_it.c

```diff
@@ -0,0 +1,81 @@
+#include "gpio_it.h"
+
+#include <stdint.h>
+#include <stdlib.h>
+
+#include "exti.h"
+#include "gpio.h"
+#include "misc.h"
+#include "nvic.h"
+#include "status.h"
+#include "stm32f0xx.h"
+#include "stm32f0xx_rcc.h"
+#include "stm32f0xx_syscfg.h"
+
+#define GPIO_NUM_PINS 16
```

Comment:
We should have device-specific headers that define these. Hardcoding values across multiple source values makes it very easy to desync.